### DISCUSSION TITLE
Any tricks for uploading code to the audio kit / port is busy

### DISCUSSION THREAD

**Discussion by jazzbiddy** [marcel-licence/esp32_midi_sampler#35] [Q&A] [answered]
I'm having a heck of a time reliably uploading any code (not really just Marcel's code) to the Audio Kit. I'll get it once every 20 tries.  I'm not really sure what makes it work when it does. sometime I press the boot button, some times I press the rst button. I've been putting a jumper on GND and IO0 in the upper right header. and I've just been plugging and unplugging the usb over and over.  Are you all having a difficult time with uploading any code also? or is this just me.

Any advise is welcome. Thanks!

**Comment by marcel-licence** [comment, marked as answer]
What kind of error did you get? Usually you don't need to press any buttons. If unsure check that IO0, IO5 and also IO12, IO13, IO14, IO15 (markes as JTAG) is not connected. Sometimes it can cause some trouble

**Reply by jazzbiddy** [reply]
At the moment, I don't really have anything connected. It's a few different errors. 

```
esptool.py v3.0-dev
Serial port COM4
Connecting......
Chip is ESP32-D0WD (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: e8:68:e7:1f:de:a8
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Warning: Could not auto-detect Flash size (FlashID=0xffffff, SizeID=0xff), defaulting to 4MB
Compressed 8192 bytes to 47...

A fatal error occurred: Timed out waiting for packet content
A fatal error occurred: Timed out waiting for packet content
```

or this one
```
esptool.py v3.0-dev
Traceback (most recent call last):
  File "esptool.py", line 3682, in <module>
Serial port COM4
  File "esptool.py", line 3675, in _main
  File "esptool.py", line 3329, in main
  File "esptool.py", line 263, in __init__
  File "site-packages\serial\__init__.py", line 88, in serial_for_url
  File "site-packages\serial\serialwin32.py", line 62, in open
serial.serialutil.SerialException: could not open port 'COM4': WindowsError(5, 'Access is denied.')
Failed to execute script esptool
the selected serial port Failed to execute script esptool
 does not exist or your board is not connected
```

or sometimes just this
`Error opening serial port 'COM4'. (Port busy)`

when I get that one, I don't have anything using the port for serial. So, like I said, It could just be my setup. I am able to upload things but it's after plugging and unplugging many times, then it just works. 

One thing I was wondering - I don't really know what the DIP switches in the middle do. I don't know if those could impact upload behavior. I don't think they should since I'm not changing them to make uploads work.  my current DIP settings are:
off,on,on,off,off

![esp32-a1s](https://user-images.githubusercontent.com/17692134/126870022-2f0d7e59-9e33-4f8b-ba75-9b2b2bd8c4fe.jpeg)

**Reply by marcel-licence** [reply]
Do you use Cura Slicer? I had a problem when using 3D printer software blocking my ports.

**Reply by jazzbiddy** [reply]
Ha. I do!  Funny that people that tinker with Arduino/ESPs most often also mess with 3d printing! 

I decided to try using VS Code for the IDE today, and it uploads without error every time so far - so it must be something with my Arduino Environment.  I've installed so many versions of things recently - different libraries and the core esp32/esp8266 board files, etc. So it must be that. I'll close this discussion for now - thanks for you help!
